src/modules/authorization/utils.ts

```diff
@@ -16,6 +16,37 @@ export function hasAuthorization(
   )
 }
 
+export function hasAuthorizationAndEnoughAllowance(
```

Comment:
I'm not sure where it's used, but, if we're storing the authorizations in the store, shouldn't this be a selector?